pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -765,6 +802,40 @@ func (gce *GCECloud) EnsureLoadBalancer(clusterName string, apiService *api.Serv
 		glog.Infof("EnsureLoadBalancer(%v(%v)): created forwarding rule, IP %s", loadBalancerName, serviceName, ipAddress)
 	}
 
+	// If requested, create DNS record with the static external ip address.
+	// If annotation removed, make sure the previous dns name is deleted.
+	// Try create//delete/update the dns name at anytime the service(LoadBalancer type) changes.
+	if gce.dnsService != nil {
+		if serviceNeedDns(apiService) {
```

Comment:
since you need the value anyway, why not just check annotations yourself:

```
if dnsName, ok :=  apiService.Annotations[serviceAnnotationLoadBalancerDns]; ok {
  ...
}
```